Update DDR4 OC Guide.md (#40)
integralfx/MemTestHelper: DDR4 OC Guide.md

   | ------ | ---- | ----- | ------- |
   | tRRDS tRRDL tFAW | 6 6 24 | 4 6 16 | 4 4 16 |
   | tWR | 16 | 12 | 10 |

>    * Minimum tFAW can be whilst improving performance is tRRDS * 4 or tRRDL * 4, depending on if tRRDS or tRRDL is lower.

Reviewer: ```suggestion
   * Minimum tFAW can be while still improving performance is tRRDS * 4 or tRRDL * 4 (whichever is lower).
```

   * You don't have to run all of the timings at one preset. You might only be able to run tRRDS tRRDL tFAW at the tight preset, but you may be able to run tWR at the extreme preset.

>    * On Intel DDR4 platforms the tWR timing does not exist and is replaced by the tWRPRE timing. This timing is the gap between the commands whilst tWR is from the last clock cycle of the write to the read command. Due to this the conversion from tWR to tWRPRE is tWR = tWRPRE - tCWL - 4. If your bios has tWRPRE it is recommended to use that as tWR isn't a real timing on Intel. However the tWR shown in something such as asrock timing configurator is the correct effective tWR. Acceptable range for tWRPRE is 23-95 on skylake mainstream platforms, and 18-159 on Rocketlake. tWR can be pushed to an effective negative timing using tWRPRE.

Reviewer: ```suggestion
   * On Intel DDR4 platforms it is recommended to use tWRPRE instead of tWR if your UEFI has it. Dropping tWRPRE by 1 will drop tWR by 1, following the rule tWR = tWRPRE - tCWL - 4.
```
Good to know but that's a long story to say "use tWRPRE instead of tWR".

   | tWTRS tWTRL | 4 12 | 4 10 | 4 8 |
   | tRTP | 12 | 10 | 8 |
   | tCWL<sup>1</sup> | tCL | tCL - 1 | tCL - 2 |

>    * Like tWR, on Intel tWTRS/L are not real timings and are replaced by tWRRD_dg and tWRRD_sg respectivly. Due to this, dropping tWRRD_dg will lower tWTRS by 1. Likewise with tWRRD_sg. In tWTRS/L readings it must be noted that the timings are read two lower then they physically, meaning a reading of 4/8 tWTRS/L in asrock timing configurator is actually 6/10. Conversion between tWTRS/L and tWRRD_sg/dg readings is tWTR = tWRRD - tCWL - 6. However this is two lower then actual, tWTR = tWWRD - tCWL - 4 is the actual effective tWTR.

Reviewer: > it must be noted that the timings are read two lower then they physically, meaning a reading of 4/8 tWTRS/L in asrock timing configurator is actually 6/10.

Source?

Author: ![image](https://user-images.githubusercontent.com/77159913/121153633-91601500-c889-11eb-9866-aba4253b5f9c.png)
tWTR is the gap between the end of the write burst and the read command.
tWRRD is the write to read command.
The length in clocks between the write command and end of the write burst is write latency (equal to tCWL without AL) plus the length of the burst, which is 4 clocks as a standard burst and 2 if burst chop is used.
So tWTR = tWRRD - tCWL - 4 for BL and tWTR = tWRRD - tCWL - 2 for BC.
BC isn't used on these platforms afaik, though that could be wrong.
And in software as shown here
![image](https://user-images.githubusercontent.com/77159913/121154084-fc115080-c889-11eb-8752-16c851e7a86c.png)
tWTR = tWRRD - tCWL - 6, which doesn't make sense at all. So it is being read 2 lower with BL or 4 lower if BC is used.

Author: ![image](https://user-images.githubusercontent.com/77159913/121154370-472b6380-c88a-11eb-93bf-735ec1a708e9.png)

8. Set `tRC = tRP + tRAS`. Increase if unstable.
   * tRC is only available on AMD and some Intel UEFIs.

>    * tRC as a timing is only aviable on AMD. On Intel this timing is not used so the effective tRC is tRAS + tRP or tRCD + tRTP + tRP, depending on which is higher. Due to this there can be many cases where stability issues are encountered due to the effective tRC being too low, as shown in the following images.

Reviewer 2: It would be more accurate to call tRC a deprecated option on Intel. I remember it being maybe x79/x99 or similar having tRC as a modifiable option which was removed with Skylake and Skylake-X. Unless it did nothing back then, in which case strange. I bring this up as tRC is commonly an option on x299 bios as a holdover iirc

Reviewer 2: This should work.

>    * tRC as a timing is only aviable on AMD. On Intel this timing is not used so the effective tRC is tRAS + tRP or tRCD + tRTP + tRP, depending on which is higher. Due to this there can be many cases where stability issues are encountered due to the effective tRC being too low, as shown in the following images.

Reviewer 2: ```suggestion
   * tRC as a timing is now deprecated on Intel. Although it is present in certain bioses, including a lot of X299 bioses, it is no longer used and is either unchangeable directly or is not used when changed. On newer Intel platforms this timing is not used so the effective tRC is `tRAS + tRP` or `tRCD + tRTP + tRP` (the one of greater value is used). Due to this there can be many cases where stability issues are encountered due to the effective tRC being too low, as shown in the following images.
```

>    * tRC as a timing is only aviable on AMD. On Intel this timing is not used so the effective tRC is tRAS + tRP or tRCD + tRTP + tRP, depending on which is higher. Due to this there can be many cases where stability issues are encountered due to the effective tRC being too low, as shown in the following images.

Reviewer 2: ```suggestion
   * tRC as a timing is now deprecated on Intel. Although it is present in certain bioses, including a lot of X299 bioses, it is no longer used and is either unchangeable directly or is not used when changed. On newer Intel platforms this timing is not used so the effective tRC is `tRAS + tRP` or `tRCD + tRTP + tRP` (the one of greater value is used). Due to this there can be many cases where stability issues are encountered due to the effective tRC being too low, as shown in the following images.
```

Reviewer 2: It would be more accurate to call tRC a deprecated option on Intel. I remember it being maybe x79/x99 or similar having tRC as a modifiable option which was removed with Skylake and Skylake-X. Unless it did nothing back then, in which case strange. I bring this up as tRC is commonly an option on x299 bios as a holdover iirc
